### DISCUSSION TITLE
Docker container ejbca can't connect to external Database

### DISCUSSION THREAD

**Discussion by srnovh** [Keyfactor/ejbca-ce#508] [General]
Hi,
we have a problem, we are building an ejbca keyfactor/ejbca-ce:7.11.0 container, when I start our container it can't connect to our external database:

**_****2024-02-15 08:41:17,560+0000 INFO [/opt/keyfactor/bin/start.sh] (process:1) Waiting for external database 'jdbc:mariadb://xxx.xxx.xxx.xxx:3000/database?characterEncoding=UTF-8?verifyServerCertificate=false&SSLMode=true&requireSSL=true' to become available.
2024-02-15 08:41:21,654+0000 INFO [org.apache.commons.beanutils.FluentPropertyBeanIntrospector] (main) Error when creating PropertyDescriptor for public final void org.apache.commons.configuration2.AbstractConfiguration.setProperty(java.lang.String,java.lang.Object)! Ignoring this property.
[ WARN] (main) Error: 1045-28000: Access denied for user ''@'ipserver' (using password: YES)
2024-02-15 08:41:21,804+0000 ERROR [org.ejbca.ui.cli.jdbc.JdbcTool] (main) (conn=2828502) Access denied for user ''@'ipserver' (using password: YES)
2024-02-15 08:41:21,804+0000 INFO [org.ejbca.ui.cli.jdbc.JdbcTool] (main) Done.**_**

we've already started another ejbca enterprise 7.12 dev instance already connected to this database, we need to test the Docker container before migrating our production environment
any help ?
we think it's about jdbc driver connexion but not sure

**Comment by primetomas** [comment]
This one is easy, it's your access configuration in the database. The database does not allow login from that ip with that db-user.

"Access denied for user ''@'ipserver'"

**Comment by srnovh** [comment]
we think that no the user is indeed authorized to this database, the same user on another ejbca ( not a docker but ejbca ee jar)  connect well

**Comment by primetomas** [comment]
I can only say that this is an error message from the database, not from EJBCA. So whatever is wrong, it's between the database jdbc connector and the database. I have never seen this error message coming from the database where it is actually not the case that there is an access right is in the configuration. You can check the database logs for more details, it will be noted there when the database sends this message back.

**Reply by srnovh** [reply]
resolved. wrong jdbc connector.
we put jar **_com.mysql.cj.jdbc.Driver_** over mariadb connector jar include in docker image
thanks
